packages/pwa/app/commerce-api/index.js

```diff
@@ -97,66 +102,80 @@ class CommerceAPI {
         const self = this
+
+        self._sdkInstances = {}
+
         Object.keys(apiConfigs).forEach((key) => {
-            const SdkClass = apiConfigs[key].api
-            self._sdkInstances = {
-                ...self._sdkInstances,
-                [key]: new Proxy(new SdkClass(this._config), {
-                    get: function(obj, prop) {
-                        if (typeof obj[prop] === 'function') {
-                            return (...args) => {
-                                const fetchOptions = args[0]
-                                const {locale, currency} = self._config
-
-                                if (fetchOptions.ignoreHooks) {
-                                    return obj[prop](...args)
-                                }
-
-                                // Inject the locale and currency to the API call via its parameters.
-                                //
-                                // NOTE: The commerce sdk isomorphic will complain if you pass parameters to
-                                // it that it doesn't expect, this is why we only add the locale and currency
-                                // to some of the API calls.
-
-                                // By default we send the locale param and don't send the currency param.
-                                const {sendLocale = true, sendCurrency = false} = apiConfigs[key]
-
-                                const includeGlobalLocale = Array.isArray(sendLocale)
-                                    ? sendLocale.includes(prop)
-                                    : !!sendLocale
-
-                                const includeGlobalCurrency = Array.isArray(sendCurrency)
-                                    ? sendCurrency.includes(prop)
-                                    : !!sendCurrency
-
-                                fetchOptions.parameters = {
-                                    ...(includeGlobalLocale ? {locale} : {}),
-                                    ...(includeGlobalCurrency ? {currency} : {}),
-                                    // Allowing individual API calls to override the global locale/currency
-                                    ...fetchOptions.parameters
-                                }
-
-                                return self.willSendRequest(prop, ...args).then((newArgs) => {
-                                    return obj[prop](...newArgs).then((res) =>
-                                        self.didReceiveResponse(res, newArgs)
-                                    )
-                                })
-                            }
-                        }
+            const SDKClass = apiConfigs[key].api
+
+            let config = this._config
+            if (apiConfigs[key].fetchOptions) {
+                config = {
+                    ...{
+                        fetchOptions: apiConfigs[key].fetchOptions
+                    },
+                    ...config
+                }
+            }
```

Comment:
This section looks like a major code change, but all I've done is introduce a new attribute `fetchOptions` to `apiConfigs`. If it is present, it is used to extend the config loaded into that SDKClass.